core/storage/exploration/gae_models.py

```diff
@@ -225,6 +211,56 @@ def delete_multi(
                 commit_log_models)
             datastore_services.put_multi(commit_log_models)
 
+    @staticmethod
+    def convert_to_valid_dict(snapshot_dict):
+        """Replace invalid fields and values in the ExplorationModel dict.
+        Some old ExplorationModels can contain fields
+        and field values that are no longer supported and would cause
+        an exception when we try to reconstitute a ExplorationModel from
+        them. We need to remove or replace these fields and values.
+
+        Args:
+            snapshot_dict: dict. The content of the model. Some fields and field
+                values might no longer exist in the ExplorationModel
+                schema.
+
+        Returns:
+            dict. The content of the model. Only valid fields and values are
+            present.
+        """
+
+        # The nodes field is moved to collection_contents dict. We
```

Comment:
What is this comment referring to?

Reply:
I added them by mistake as I copy-pasted some existing code.

Reply:
Removed the comments. Done.